README.md

```diff
@@ -271,69 +277,65 @@ python -m unittest -bv
-    changes to your PR.
-* [gersemi][9]: `gersemi -i -l 100 --list-expansion favour-expansion CMakeLists.txt`
-  * This formats CMakeLists.txt. You should review and add any changes to your PR.
-
-Note that `docformatter` should be run before `black` to give Black the
-[last][12].
-
-Additionally, the following tools can be useful during development. However,
-they cannot be installed using `pip`. Developers need to install them using
-other package management tools such as `apt-get`:
-
-* [clang-tidy][8]: `clang-tidy --extra-arg=-std=c++17 PATH_TO_THE_FILE`
-  * This static analysis tool catches improper coding behaviors based on the
-    rules specified in `.clang-tidy`, and sends suggestions corresponding to
-    each warning. Developers should manually review all the warnings and try
-    with their best effort to fix the reasonable ones.
+## Adding files
+Certain file types need to be added to our linting rules manually:
+
+- **CMake**. If adding a CMake file, add it (or its parent directory) as an argument to the
+  `gersemi` command in [lint-tasks.yaml](lint-tasks.yaml).
+  * If adding a directory, the file must be named `CMakeLists.txt` or use the `.cmake` extension.
+- **YAML**. If adding a YAML file (regardless of its extension), add it as an argument to the
+  `yamllint` command in [lint-tasks.yaml](lint-tasks.yaml).
+
+## Linting
+Before submitting a pull request, ensure you’ve run the linting commands below and either fixed any
+violations or suppressed the warning.
+
+To run all linting checks:
+```shell
+task lint:check
+```
+
+To run all linting checks AND automatically fix any fixable issues:
+```shell
+task lint:fix
+```
+
+### Running specific linters
+The commands above run all linting checks, but for performance you may want to run a subset (e.g.,
+if you only changed C++ files, you don't need to run the YAML linting checks) using one of the tasks
+in the table below.
+
+| Task                    | Description                                              |
+|-------------------------|----------------------------------------------------------|
+| `lint:cmake-check`      | Runs the CMake linters.                                  |
+| `lint:cmake-fix`        | Runs the CMake linters and fixes any violations.         |
+| `lint:cpp-check`        | Runs the C++ linters (formatters and static analyzers).  |
+| `lint:cpp-fix`          | Runs the C++ linters and fixes some violations.          |
+| `lint:cpp-format-check` | Runs the C++ formatters.                                 |
+| `lint:cpp-format-fix`   | Runs the C++ formatters and fixes some violations.       |
+| `lint:cpp-static-check` | Runs the C++ static analyzers.                           |
+| `lint:cpp-static-fix`   | Runs the C++ static analyzers and fixes some violations. |
+| `lint:yml-check`        | Runs the YAML linters.                                   |
+| `lint:yml-fix`          | Runs the YAML linters and fixes some violations.         |
```

Comment:
- `lint:cpp-static-check/fix` are not supported yet, right?
- We should also mention `lint:py-check/fix`

Reply:
lint:cpp-static-check/fix is actually supported. you can run it directly, it is just included in the lint:fix or lint:check flow.
I thought it would be helpful when you start to fix clang-tidy errors. Do you want me to change this behavior?

Reply:
I think let's remove it for now since it's not automated. Let me turn it on once all warnings are fixed

Reply:
understood